### DISCUSSION TITLE
Add UI and API to download the full event payload that triggered an action

### DISCUSSION THREAD

**Discussion by gr2m** [community/community#12622] [Actions]
When an action fails, the next step usually is to replicate the same problem locally. But if I didn't add an explicit step to log out an event payload, there is no way to retrieve the event payload that triggered the workflow build.

Being able to programmatically retrieve the event payload would make debugging workflow errors so much easier. It would be nice to add an ability both in the UI (similar to what GitHub Apps have today) and an API for programmatic access.

As an alternative approach, I've been logging out the event payload with a simple job that I add to all workflows:

```yml
  debug:
    runs-on: ubuntu-latest
    steps:
      - name: "log event payload for debugging"
        run: cat $GITHUB_EVENT_PATH
```

But sometimes I forget or it's not my workflow file, and when a problem occurred it's too late anyway, there is no way to retrieve the event that triggered it.

As I'm looking into programmatic access I'm looking into uploading the event payload as an workflow build artifact. But it really makes me think this should be a built-in feature

**Comment by ethomson** [comment]
Thanks for the feedback @gr2m - I'll have to give this one some thought, but it seems like a reasonable request.

**Reply by gr2m** [reply]
Big bonus points if we could access any outputs as well, at least the ones that are used by other steps. With the event and the outputs being accessible, we could replicate the environment for local debugging with only a URL to a failed workflow build as input
